djproxy/views.py

```diff
@@ -135,6 +135,10 @@ def proxy(self):
         headers['X-Forwarded-Host'] = self.request.get_host()
         headers['X-Forwarded-For'] = self.xff
 
+        for prefix, url in self.reverse_urls:
+            if url == self.proxy_url and self.request.path.startswith(prefix):
```

Comment:
So, the behavior you'd see with this implemented as is:

Going to /yahoo/ would be good. The prefix would be there. Going to /yahoo/yay/ would be broken, the prefix wouldn't be there.